weld/llvm.rs

```diff
@@ -125,78 +125,152 @@ impl LlvmGenerator {
 
     /// Return all the code generated so far.
     pub fn result(&mut self) -> String {
-        format!("; PRELUDE:\n\n{}\n; BODY:\n\n{}",
-                self.prelude_code.result(),
-                self.body_code.result())
+        format!(
+            "; PRELUDE:\n\n{}\n; BODY:\n\n{}",
+            self.prelude_code.result(),
+            self.body_code.result()
+        )
+    }
+
+    fn unary_op(&mut self, ctx: &mut FunctionContext, func: &SirFunction,
```

Comment:
This is a little weird; maybe just pass the `UnaryOpKind` into this function and produce the `llvm_func` here? Otherwise if we add other things which don't have `llvm` in the function name, it will fail and generate `erf`